tests/src/PatternAssertion/PaginationPatternAssert.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_bootstrap_theme\PatternAssertion;
+
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Assertions for the pagination pattern.
+ */
+class PaginationPatternAssert extends BasePatternAssert {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAssertions(string $variant): array {
+    return [
+      'links' => [
+        [$this, 'assertLinks'],
+      ],
+      'alignment' => [
+        [$this, 'assertAlignment'],
+      ],
+      'size' => [
+        [$this, 'assertSize'],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function assertBaseElements(string $html, string $variant): void {
+    $this->assertCounts([
+      'nav' => 1,
+      'ul' => 1,
+    ], new Crawler($html));
+  }
+
+  /**
+   * Asserts the pager links.
+   *
+   * @param array[] $expected
+   *   Expected pager links.
+   *   Each link is an array, with the following keys:
+   *     - 'url': The link destination.
+   *     - 'label': The label, if exists.
+   *     - 'icon': The name of the icon.
+   *     - 'active': Whether the item has an 'active' class.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The crawler.
+   */
+  protected function assertLinks(array $expected, Crawler $crawler): void {
+    $expected_bcl_icon_url = '/' . \Drupal::service('extension.list.theme')->getPath('oe_bootstrap_theme') . '/assets/icons/bcl-default-icons.svg';
+
+    $actual_links_data = array_map(
+      function (\DOMElement $actual_li_element) use ($expected_bcl_icon_url) {
+        $li = new Crawler($actual_li_element);
+        $link = $li->filter('a');
+        $actual_link_data = [];
+        $actual_link_data['url'] = $link->attr('href');
+        $use = $li->filter('a > svg > use');
+        if ($use->count()) {
+          // This is an icon.
+          $icon_url = $use->attr('xlink:href');
+          [$actual_bcl_icon_url, $actual_link_data['icon']] = explode('#', $icon_url, 2) + [NULL];
```

Comment:
Actually the `+ [NULL]` is not just hard to read, it is wrong / insufficient.
The goal is to always have an array with two elements to avoid "Warning: Undefined array key 1".
To achieve this we need `+ [NULL, NULL]`, not just `+ [NULL]`.
https://3v4l.org/aATpQ

Much better would be if PHP would allow a default value in the array unpacking.
https://3v4l.org/JRgC2
But this is not the case.

I am going to drop the array unpacking and use separate assignments instead.
(and then drop the entire thing to go for single assert)